lib/matplotlib/axes/_axes.py

```diff
@@ -1522,7 +1522,8 @@ def plot(self, *args, scalex=True, scaley=True, data=None, **kwargs):
         ----------------
         scalex, scaley : bool, default: True
             These parameters determine if the view limits are adapted to the
-            data limits. The values are passed on to `autoscale_view`.
+            data limits. The values are passed on to
+            `~matplotlib.axes.Axes.autoscale_view`.
```

Comment:
https://matplotlib.org/devdocs/devel/documenting_mpl.html#referencing-types

```suggestion
            `~.Axes.autoscale_view`.
```

There were times when this caused confusion (=doc build warnings) with the subclasses `mpl_toolkits.axes_grid1.mpl_axes.Axes` / `mpl_toolkits.axisartist.mpl_axes.Axes`. However, we do have working `` `~.Axes.*` ``in our docs. Either we have some priority setting there that I'm not aware of, or it works as long as the other Axes classes do not document the same method.

If you want to be on the safe side for Axes
```suggestion
            `~.axes.Axes.autoscale_view`.
```

Just counted: We have 150 `~.axes.Axes*` and 178 `~.Axes.*` so :shrug:.
Anyway leaving out the `matplotlib` part makes it more readable.

Reply:
Same for the following 2 cases.

Reply:
Just to be clear about the background. The issue happens for the pyplot version of the documentation. It may be that one can remove `matplotlib`, but probably not more than that. Just wanted to be on the safe side and it should render just as `autoscale_view`. But not so clear in the source, no.

There are more similar cases where the pyplot documentation is not linked properly.

Reply:
This is the relevant section of sphinx https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#role-py-obj

>Normally, names in these roles are searched first without any further qualification, then with the current module name prepended, then with the current module and class name (if any) prepended. If you prefix the name with a dot, this order is reversed. For example, in the documentation of Python’s [codecs](https://docs.python.org/3/library/codecs.html#module-codecs) module, :py:func:`open` always refers to the built-in function, while :py:func:`.open` refers to [codecs.open()](https://docs.python.org/3/library/codecs.html#codecs.open).
>
> A similar heuristic is used to determine whether the name is an attribute of the currently documented class.
>
> Also, if the name is prefixed with a dot, and no exact match is found, the target is taken as a suffix and all object names with that suffix are searched. For example, :py:meth:`.TarFile.close` references the tarfile.TarFile.close() function, even if the current module is not tarfile. Since this can get ambiguous, if there is more than one possible match, you will get a warning from Sphinx.
>
> Note that you can combine the ~ and . prefixes: :py:meth:`~.TarFile.close` will reference the tarfile.TarFile.close() method, but the visible link caption will only be close().

It get's funny when docstrings are reused either through our copy mechanism in pyplot or through inheritance in another module.

`autoscale_view` works in docstrings of `_axes.py` because it's in there, but it does not work in pyplot docstrings becaue there is no `plt.autoscale_view`.

Since it's hard to keep track of what is referenced and reused where, we usually use the `.` mechanism, in particular in Axes methods.